zebra-state/src/service.rs

```diff
@@ -797,6 +797,84 @@ impl ReadStateService {
     fn latest_non_finalized_state(&self) -> NonFinalizedState {
         self.non_finalized_state_receiver.cloned_watch_data()
     }
+
+    /// Check the contextual validity of a block in the best chain
+    #[cfg(feature = "getblocktemplate-rpcs")]
+    #[allow(clippy::unwrap_in_result)]
+    #[tracing::instrument(level = "debug", skip_all)]
+    fn check_best_chain_contextual_validity(
+        &self,
+        block: PreparedBlock,
+    ) -> Result<(), crate::ValidateContextError> {
+        if let Some(best_chain) = self.latest_non_finalized_state().best_chain() {
+            let best_chain_latest_non_finalized_state = NonFinalizedState::build(self.network)
+                .insert_chain(Arc::clone(best_chain))
```

Comment:
qualifying the `Arc::clone` so it's obvious that nothing's being copied without looking at the type.